include/range/v3/view/iota.hpp

```diff
@@ -163,6 +163,25 @@ namespace ranges
                     :  static_cast<D>(v0 - v1);
             }
 
+            template<typename Val, CONCEPT_REQUIRES_(SignedIntegral<Val>())>
+            RANGES_CXX14_CONSTEXPR
+            iota_difference_t<Val> ints_distance_(Val from, Val to) noexcept {
+                using D = iota_difference_t<Val>;
+                RANGES_EXPECT(from <= to);
+                RANGES_EXPECT((static_cast<D>(to) - static_cast<D>(from))
+                              < std::numeric_limits<D>::max());
```

Comment:
So this check uses `<` instead of `<=` to catch when the range size which is `(to - from) + 1` for closed ranges would overflow. That is, if `to - from == max` everything is still fine, but when one attempts to compute the size that will overflow for _unsigned integers_.

However, we are dealing with signed integers here which have an unsigned range size that is at least as wide, so doing that should never overflow, so this check is moot and can be removed. 

> If it's an attempt to guard against to - from overflowing D, which is commendable, it's not working.

I don't think that can ever work. If `to - from` overflows its UB, I don't see a way to detect this.